packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -136,7 +136,10 @@ export const addEdge = (
 
 	// Check if type is compatible
 	if (sourceOutputPort.value === null) return;
-	if (sourceOutputPort.type !== targetInputPort.type) return;
+
+	const allowedTypes = targetInputPort.type.split('|');
```

Comment:
Should we think about making those `inputPort.type` an array instead?

Reply:
I have not thought about this option. We can have a chat about it along with other potential improvement like using concrete classes instead of types.